### ISSUE TITLE
Allow to disable default resolver

### ISSUE THREAD

**Issue by bertho-zero** [import-js/eslint-plugin-import#3095]
By default eslint merges the config, making it impossible to use a custom resolver with option without the default resolver node being added.

###  Try 1

```js
settings: {
  'import/resolver': {
    [path.resolve(__dirname, './resolver')]: { /* some options */ },
  },
},
```

Then the following resolvers are used:
```js
{
  '/home/.../eslint-config/resolver': {},
  node: { extensions: [ '.js', '.jsx', '.json' ] }
}
```

###  Try 2:

```js
settings: {
  'import/resolver': {
    [path.resolve(__dirname, './resolver')]: { /* some options */ },
    node: false,
  },
},
```

Then the following resolvers are used:
```js
{
  '/home/.../eslint-config/resolver': {},
  node: false,
}
```

And the node resolver is still used, same with null, undefined or other

###  Try 3:

I can only disable resolver node by doing this, but I can't pass any options:

```js
settings: {
  'import/resolver': path.resolve(__dirname, './resolver'),
},
```

---

I propose as a solution not to use a resolver if its configuration is equal to false.

**Comment by ljharb** [comment]
I would definitely expect `false` to work here.
